### PR TITLE
Update package baseline

### PR DESCRIPTION
**Pull request by wtgodbe**
Contributes to https://github.com/aspnet/AspNetCore/issues/12702

The update to `Baseline.xml` is manual - I added all packages that AspNetCore shipped in 3.0 GA, but picked only 1 version of the shared framework package (`Microsoft.AspNetCore.App.runtime-winx64`). The update to `Basline.Designer.props` was done by running the BaselineGenerator project.

Replaces https://github.com/aspnet/AspNetCore/pull/14437

### CODE REVIEW

**Comment by wtgodbe** [comment]
Interesting:

> F:\workspace\_work\1\s\src\Framework\ref\Microsoft.AspNetCore.App.Ref.csproj(182,5): error MSB3073: The command "C:\Windows\System32\tar.exe -czf F:\workspace\_work\1\s\artifacts\installers\Release\aspnetcore-targeting-pack-3.0.0.tar.gz ." exited with code 9009.

Implies that `C:\Windows\System32\tar.exe` _does_ exist, given https://github.com/aspnet/AspNetCore/pull/14535/commits/4165a0282c9207adbff31e5b18fda8469ef2a8ff.

File: src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj [hunk 1]
```diff
@@ -154,6 +154,14 @@ This package is an internal implementation of the .NET Core SDK and is not meant
     <CreateDirectory Include="$(LocalInstallationOutputPath)" />
   </ItemGroup>
 
+  <!-- Workaround until https://github.com/dotnet/core-eng/issues/7970 is resolved -->
+  <PropertyGroup>
+    <TarCommand Condition="'$(OS)' == 'Windows_NT'">C:\Windows\System32\tar.exe</TarCommand>
+    <TarCommand Condition="'$(OS)' == 'Windows_NT' and !Exists($(TarCommand))">'C:\Program Files\Git\usr\bin\tar.exe'</TarCommand>
+    <TarCommand Condition="'$(OS)' == 'Windows_NT' and !Exists($(TarCommand))">'C:\Program Files (x86)\Git\usr\bin\tar.exe'</TarCommand>
+    <TarCommand Condition="'$(OS)' != 'Windows_NT'">tar</TarCommand>
```
Comment (dougbu) [inline src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj]:
```suggestion
    <TarCommand>tar</TarCommand>
    <TarCommand Condition="'$(OS)' == 'Windows_NT'">C:\Windows\System32\tar.exe</TarCommand>
    <TarCommand Condition="'$(OS)' == 'Windows_NT' and !Exists('$(TarCommand)')">C:\Program Files\Git\usr\bin\tar.exe</TarCommand>
    <TarCommand Condition="'$(OS)' == 'Windows_NT' and !Exists('$(TarCommand)')">C:\Program Files (x86)\Git\usr\bin\tar.exe</TarCommand>
```

Single quotes aren't going to work w/ `cmd` but are useful when calling `Exists(...)`. And, this simplifies things a bit.

File: src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj [hunk 2]
```diff
@@ -175,7 +183,7 @@ This package is an internal implementation of the .NET Core SDK and is not meant
       Overwrite="true" />
     <!-- Requires Windows 10 version 1803 or newer -->
     <Exec
-      Command="tar -czf $(TarArchiveOutputPath) ."
+      Command="$(TarCommand) -czf $(TarArchiveOutputPath) ."
```
Comment (dougbu) [inline src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj]:
```suggestion
      Command="&quot;$(TarCommand)&quot; -czf $(TarArchiveOutputPath) ."
```

Quotes are also great in `<Exec />` 😺

File: src/Framework/src/Microsoft.AspNetCore.App.Runtime.csproj [hunk 3]
```diff
@@ -363,6 +363,15 @@ This package is an internal implementation of the .NET Core SDK and is not meant
   Targets related to creating .zip/.tar.gz
   #########################################
   -->
+
+  <!-- Workaround until https://github.com/dotnet/core-eng/issues/7970 is resolved -->
+  <PropertyGroup>
+    <TarCommand Condition="'$(OS)' == 'Windows_NT'">C:\Windows\System32\tar.exe</TarCommand>
+    <TarCommand Condition="'$(OS)' == 'Windows_NT' and !Exists($(TarCommand))">'C:\Program Files\Git\usr\bin\tar.exe'</TarCommand>
+    <TarCommand Condition="'$(OS)' == 'Windows_NT' and !Exists($(TarCommand))">'C:\Program Files (x86)\Git\usr\bin\tar.exe'</TarCommand>
+    <TarCommand Condition="'$(OS)' != 'Windows_NT'">tar</TarCommand>
```
Comment (dougbu) [inline src/Framework/src/Microsoft.AspNetCore.App.Runtime.csproj]: Same deal as the other project

**Review by dougbu — APPROVED** [review]
Once things work, go for it

Please file the follow-up bug about improving the baseline generator so that we can include the shared framework in the baseline

**Comment by dougbu** [comment]
Not sure what I hit to close this. Entirely unintended.

**Comment by dougbu** [comment]
Previous build got `tar` from `git` but the single-quotes messed it up:
```
The filename, directory name, or volume label syntax is incorrect.
F:\workspace\_work\1\s\src\Framework\ref\Microsoft.AspNetCore.App.Ref.csproj(185,5): error MSB3073: The command "'C:\Program Files (x86)\Git\usr\bin\tar.exe' -czf F:\workspace\_work\1\s\artifacts\installers\Release\aspnetcore-targeting-pack-3.0.0.tar.gz ." exited with code 123.
```

BTW one of @alexperovich's theories was that the agents were inconsistent. He found `tar` on `$env:Path` in at least some of the agents.

**Comment by dougbu** [comment]
FYI I stole a variant of your `tar` workaround in #14404

**Pull request merged**